File: python/sns_autopilot/publish/adapters/webhook.py
"""n8n / Make / Zapier / Slack 등으로 통째로 넘깁니다."""
from __future__ import annotations

import os
from typing import Any

import requests

from ..compose import LIMITS, compose_text
from .base import Adapter


class WebhookAdapter(Adapter):
    name = "webhook"
    required_env = ("WEBHOOK_URL",)

    def send(self, post: dict, assets: dict, ctx: dict) -> dict[str, Any]:
        payload = {
            "runId": ctx["run_id"],
            "platform": post["platform"],
            "variant": post["variant"],
            "title": post.get("title") or None,
            "text": compose_text(post, limit=LIMITS.get(post["platform"], 0)),
            "hashtags": post.get("hashtags", []),
            "firstComment": post.get("firstComment") or None,
            "assets": assets,
        }
        response = requests.post(os.environ["WEBHOOK_URL"], json=payload, timeout=30)
        if not response.ok:
            raise RuntimeError(f"webhook {response.status_code}: {response.text[:200]}")
        return {"ok": True, "target": self.name, "status": response.status_code}

File: src/publish/adapters/webhook.js
import { composeText, LIMITS } from "../compose.js";

/** n8n / Make / Zapier / Slack 등으로 통째로 넘깁니다. */
export const webhook = {
  name: "webhook",
  configured: () => Boolean(process.env.WEBHOOK_URL),
  async send({ post, assets, runId }) {
    const body = {
      runId,
      platform: post.platform,
      variant: post.variant,
      title: post.title || undefined,
      text: composeText(post, { limit: LIMITS[post.platform] ?? 0 }),
      hashtags: post.hashtags,
      firstComment: post.firstComment || undefined,
      assets,
    };
    const res = await fetch(process.env.WEBHOOK_URL, {
      method: "POST",
      headers: { "content-type": "application/json" },
      body: JSON.stringify(body),
    });
    if (!res.ok) throw new Error(`webhook ${res.status}: ${(await res.text()).slice(0, 200)}`);
    return { ok: true, target: "webhook", status: res.status };
  },
};
